File: typescript/phone-number/phone-number.ts
class PhoneNumber {
  private _phoneNumber: string

  constructor(phoneNumber: string) {
    this._phoneNumber = phoneNumber.replace(/(\)|\(|\s|\.|-)/g, "")
    if (this._phoneNumber.length === 11 && this._phoneNumber[0] === "1") {
      this._phoneNumber = this._phoneNumber.slice(1, 11)
    }
  }

  number(): string | undefined {
    if (this._phoneNumber.length < 10 || this._phoneNumber.length > 10) {
      return undefined
    }

    return this._phoneNumber.split("").filter((c) => isNaN(parseInt(c, 10))).length > 0
      ? undefined
      : this._phoneNumber
  }
}

export default PhoneNumber

File: csharp/phone-number/PhoneNumber.cs
using System;
using System.Text.RegularExpressions;

public class PhoneNumber
{
    private static readonly Regex Digits = new Regex(@"[^\d]");

    public string Number { get; }

    public string AreaCode { get; }

    public PhoneNumber(string phoneNumber)
    {
        Number = ParsePhoneNumber(phoneNumber);
        AreaCode = Number.Substring(0, 3);
    }

    public override string ToString() => 
        $"({AreaCode}) {Number.Substring(3, 3)}-{Number.Substring(6)}";

    private static string ParsePhoneNumber(string phoneNumber)
    {
        phoneNumber = Digits.Replace(phoneNumber, "");

        if (phoneNumber.Length != 10)
            return IsUSPhoneNumber(phoneNumber) ? phoneNumber.Substring(1) : "0000000000";

        return phoneNumber;
    }

    private static bool IsUSPhoneNumber(string phoneNumber) =>
        phoneNumber.Length == 11 && phoneNumber.StartsWith("1");
}